server/uptime-calculator.js

```diff
@@ -741,20 +750,36 @@ class UptimeCalculator {
     }
 
     /**
-     * Get the uptime data by duration
-     * @param {'24h'|'30d'|'1y'} duration Only accept 24h, 30d, 1y
+     * Get the uptime data for given duration.
+     * @param {string} duration  A string with a number and a unit (m,h,d,w,M,y), such as 24h, 30d, 1y.
      * @returns {UptimeDataResult} UptimeDataResult
-     * @throws {Error} Invalid duration
+     * @throws {Error} Invalid duration / Unsupported unit
      */
     getDataByDuration(duration) {
-        if (duration === "24h") {
-            return this.get24Hour();
-        } else if (duration === "30d") {
-            return this.get30Day();
-        } else if (duration === "1y") {
-            return this.get1Year();
-        } else {
-            throw new Error("Invalid duration");
+        const durationNumStr = duration.slice(0, -1);
+
+        if (!isNumeric(durationNumStr)) {
+            throw new Error(`Invalid duration: ${duration}`);
+        }
+        const num = Number(durationNumStr);
+        const unit = duration.slice(-1);
+
+        switch (unit) {
+            case durationUnits.MINUTE:
+                return this.getData(num, "minute");
+            case durationUnits.HOUR:
+                return this.getData(num, "hour");
+            case durationUnits.DAY:
+                return this.getData(num, "day");
+            case durationUnits.WEEK:
+                return this.getData(7 * num, "day");
+            case durationUnits.MONTH:
+                return this.getData(30 * num, "day");
+            case durationUnits.YEAR:
+                return this.getData(365 * num, "day");
```

Comment:
```suggestion
            case "M":
                return this.getData(num, "minute");
            case "h":
                return this.getData(num, "hour");
            case "d":
                return this.getData(num, "day");
            case "w":
                return this.getData(7 * num, "day");
            case "m":
                return this.getData(30 * num, "day");
            case "y":
                return this.getData(365 * num, "day");
```

Reply:
As usal, `m` is `minute` and `M` is `month`.
https://day.js.org/docs/en/durations/creating#list-of-all-available-units